packages/react-query/src/useBaseQuery.ts

```diff
@@ -31,6 +31,14 @@ export function useBaseQuery<
   Observer: typeof QueryObserver,
   queryClient?: QueryClient,
 ) {
+  if (process.env.NODE_ENV !== 'production') {
+    if (!isPlainObject(options)) {
```

Comment:
I think we should inline the check here to avoid making the `isPlainObject` function exported from the core. A simple check like this should suffice:

```
!!options && !Array.isArray(options) && typeof options === 'object'
```

we could also leave out the Array check because that was never a supported syntax in the first place :)

Reply:
Yep I agree. I did this at first and then saw `isPainObject` and I gave it a chance :).
Will change this later.

Thx.

Reply:
yep 👍

Reply:
pushed